client/data/users/with-users.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import useUsersQuery from './use-users-query';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+const withUsers = createHigherOrderComponent(
+	( Wrapped, fetchOptions = {}, queryOptions = {} ) => ( props ) => {
```

Comment:
Not sure but do we ever need to make `fetchOptions` and `queryOptions` available to the outside?

Reply:
Perhaps, yes. Imagine if we wanted to search by a term or provide a limit, then those would be useful. But because we're not using those yet, I'm happy to remove them if you'd prefer.

Reply:
That makes sense although I think constructing those objects should be part of the HOC or in general the component which use the query hook. We can iterate later though, no need to remove those now.